pkg/providers/pricing/pricing.go

```diff
@@ -225,85 +231,97 @@ func (p *DefaultProvider) UpdateOnDemandPricing(ctx context.Context) error {
 	return nil
 }
 
-func (p *DefaultProvider) fetchOnDemandPricing(ctx context.Context, additionalFilters ...*pricing.Filter) (map[string]float64, error) {
+func (p *DefaultProvider) fetchOnDemandPricing(ctx context.Context, additionalFilters ...pricingtypes.Filter) (map[string]float64, error) {
 	prices := map[string]float64{}
-	filters := append([]*pricing.Filter{
-		{
-			Field: aws.String("regionCode"),
-			Type:  aws.String("TERM_MATCH"),
-			Value: aws.String(p.region),
-		},
-		{
-			Field: aws.String("serviceCode"),
-			Type:  aws.String("TERM_MATCH"),
-			Value: aws.String("AmazonEC2"),
-		},
-		{
-			Field: aws.String("preInstalledSw"),
-			Type:  aws.String("TERM_MATCH"),
-			Value: aws.String("NA"),
-		},
-		{
-			Field: aws.String("operatingSystem"),
-			Type:  aws.String("TERM_MATCH"),
-			Value: aws.String("Linux"),
-		},
-		{
-			Field: aws.String("capacitystatus"),
-			Type:  aws.String("TERM_MATCH"),
-			Value: aws.String("Used"),
-		},
-		{
-			Field: aws.String("marketoption"),
-			Type:  aws.String("TERM_MATCH"),
-			Value: aws.String("OnDemand"),
-		}},
-		additionalFilters...)
-
-	err := p.pricing.GetProductsPagesWithContext(
-		ctx,
-		&pricing.GetProductsInput{
-			Filters:     filters,
-			ServiceCode: aws.String("AmazonEC2"),
-		},
-		p.onDemandPage(ctx, prices),
-	)
-	if err != nil {
-		return nil, err
+	filters := make([]pricingtypes.Filter, 0, len(additionalFilters)+6)
+	filters = append(filters, pricingtypes.Filter{
+		Field: aws.String("regionCode"),
+		Type:  "TERM_MATCH",
+		Value: aws.String(p.region),
+	})
+	filters = append(filters, pricingtypes.Filter{
+		Field: aws.String("serviceCode"),
+		Type:  "TERM_MATCH",
+		Value: aws.String("AmazonEC2"),
+	})
+	filters = append(filters, pricingtypes.Filter{
+		Field: aws.String("preInstalledSw"),
+		Type:  "TERM_MATCH",
+		Value: aws.String("NA"),
+	})
+	filters = append(filters, pricingtypes.Filter{
+		Field: aws.String("operatingSystem"),
+		Type:  "TERM_MATCH",
+		Value: aws.String("Linux"),
+	})
+	filters = append(filters, pricingtypes.Filter{
+		Field: aws.String("capacitystatus"),
+		Type:  "TERM_MATCH",
+		Value: aws.String("Used"),
+	})
+	filters = append(filters, pricingtypes.Filter{
+		Field: aws.String("marketoption"),
+		Type:  "TERM_MATCH",
+		Value: aws.String("OnDemand"),
+	})
+
+	for _, filter := range additionalFilters {
+		filters = append(filters, pricingtypes.Filter{
+			Field: filter.Field,
+			Type:  filter.Type,
+			Value: filter.Value,
+		})
+	}
+
+	input := &pricing.GetProductsInput{
+		Filters:     filters,
+		ServiceCode: aws.String("AmazonEC2"),
+	}
+
+	paginator := pricing.NewGetProductsPaginator(p.pricing, input)
+	for paginator.HasMorePages() {
+		output, err := paginator.NextPage(ctx)
+
+		if err != nil {
+			return nil, fmt.Errorf("getting pricing data, %w", err)
+		}
+		prices = lo.Assign(prices, p.onDemandPage(ctx, output))
```

Comment:
I'm putting a pin in this for right now (I'll need to come back to this) but we should double-check this since this is a behavior change